wallet/src/components/IbcTransfer.tsx

```diff
@@ -0,0 +1,491 @@
+/* eslint-disable import/no-extraneous-dependencies */
+import { useEffect, useMemo, useState } from 'react';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { parseAsValue } from '@agoric/ui-components';
+import { withApplicationContext } from '../contexts/Application';
+import Petname from './Petname';
+import { ibcAssets } from '../util/ibc-assets';
+import { stringifyPurseValue } from '@agoric/ui-components';
+import ArrowDownward from '@mui/icons-material/ArrowDownward';
+import { Box } from '@mui/system';
+import { fromBech32 } from '@cosmjs/encoding';
+import { queryBankBalances } from '../util/queryBankBalances';
+import type { PurseInfo } from '../service/Offers';
+import type { KeplrUtils } from '../contexts/Provider';
+import { SigningStargateClient } from '@cosmjs/stargate';
+import { CircularProgress, Link, Snackbar, Typography } from '@mui/material';
+
+export enum IbcDirection {
+  Deposit,
+  Withdrawal,
+}
+
+const titlePreposition = {
+  [IbcDirection.Deposit]: 'from',
+  [IbcDirection.Withdrawal]: 'to',
+};
+
+const agoricAddressLabel = {
+  [IbcDirection.Deposit]: 'To',
+  [IbcDirection.Withdrawal]: 'From',
+};
+
+const remoteChainAddressLabel = {
+  [IbcDirection.Deposit]: 'From',
+  [IbcDirection.Withdrawal]: 'To',
+};
+
+interface Params {
+  purse?: PurseInfo;
+  direction: IbcDirection;
+  handleClose: () => void;
+  keplrConnection: KeplrUtils;
+}
+
+const secondsUntilTimeout = 300;
```

Comment:
timeout of what? still reading...

Reply:
Just refactored this into its own util file so hopefully it's clearer now